core/src/main/java/org/web3j/protocol/core/filters/Filter.java

```diff
@@ -136,12 +139,15 @@ private void pollFilter(EthFilter ethFilter) {
         }
         if (ethLog.hasError()) {
             Error error = ethLog.getError();
+            String message = error.getMessage();
             switch (error.getCode()) {
                 case RpcErrors.FILTER_NOT_FOUND:
                     reinstallFilter();
                     break;
                 default:
-                    throwException(error);
+                    if (Pattern.compile(FILTER_NOT_FOUND_PATTERN).matcher(message).find())
+                        reinstallFilter();
+                    else throwException(error);
```

Comment:
```suggestion
                     Pattern.compile(FILTER_NOT_FOUND_PATTERN).matcher(message).find())? 
                    reinstallFilter():throwException(error);
```

Also run build locally before push

Reply:
As I pointed out [above](https://github.com/web3j/web3j/pull/1869#discussion_r1121893565) it is not possible to use ternary operator, build will fail.
Also, I should have built locally with _./gradlew check_ and I've got no errors with last commit.